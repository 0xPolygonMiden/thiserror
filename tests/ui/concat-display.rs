use thiserror::Error;

macro_rules! error_type {
    ($name:ident, $what:expr) => {
        #[derive(Error, Debug)]
        #[error(concat!("invalid ", $what))]
        pub struct $name;
    };
}

error_type!(Error, "foo");

fn main() {}
