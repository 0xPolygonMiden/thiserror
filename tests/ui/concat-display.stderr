error: expected literal
  --> $DIR/concat-display.rs:6:17
   |
6  |         #[error(concat!("invalid ", $what))]
   |                 ^^^^^^
...
11 | error_type!(Error, "foo");
   | -------------------------- in this macro invocation
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
